swan-lake/resources/featured-scenarios/write-a-restful-api-with-ballerina.md

```diff
@@ -127,103 +187,77 @@ service /covid/status on new http:Listener(9000) {
 
+![Create POST resource](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/create-post-resource.gif)
+
+Implement the logic of this `POST` resource function with the code below.
 
 ```ballerina
-resource function post countries(@http:Payload CovidEntry[] covidEntries)
-                                    returns CovidEntry[]|ConflictingIsoCodesError {
+resource function post countries(CovidEntry[] covidEntries)
+        returns CovidEntry[]|IsoCodeConflict {
 
     string[] conflictingISOs = from CovidEntry covidEntry in covidEntries
         where covidTable.hasKey(covidEntry.iso_code)
         select covidEntry.iso_code;
 
     if conflictingISOs.length() > 0 {
         return {
-            body: {
-                errmsg: string:'join(" ", "Conflicting ISO Codes:", ...conflictingISOs)
-            }
+            body: string:'join(" ", "Conflicting ISO Codes:", ...conflictingISOs)
         };
-    } else {
-        covidEntries.forEach(covdiEntry => covidTable.add(covdiEntry));
-        return covidEntries;
     }
+
+    covidEntries.forEach(covidEntry => covidTable.add(covidEntry));
+    return covidEntries;
 }
 ```
 
 In this code:
 
-- It is chosen to either accept the entire payload or send back an error. Copying this straightway results in an error, which is expected as the `ConflictingIsoCodesError` type is not defined yet.
-- This resource has a resource argument named `covidEntries` annotated with `@http:Payload`. This means the resource is expecting a payload with the `CovideEntry[]` type. There are two types of records `CovideEntry[]` and `ConflictingIsoCodesError` that will be used as the return values.
-
-#### Define the error records
-
-To define the error records, add the code below to the API template file (i.e., `service.bal`).
-
-```ballerina
-public type ConflictingIsoCodesError record {|
-    *http:Conflict;
-    ErrorMsg body;
-|};
-
-public type ErrorMsg record {|
-    string errmsg;
-|};
-```
-
-In this code:
-- Ballerina uses `*http:Conflict` to denote that one type is a subtype of another. In this case, `ConflictingIsoCodesError` is a subtype of `http:Conflict`.
-- The body of the response is of type `ErrorMsg`, which simply has a string field named `errmsg`. Based on the need, users can have any data type for their response body.
-- Ballerina has a defined set of types for each HTTP status code. This allows you to write services in a type-oriented way, which in turn is helpful when it comes to tooling and generating OpenAPI specifications for HTTP services. 
+- It is chosen to accept the entire payload or send back an error.
+- This resource has an argument named `covidEntries`, which means the resource expects a payload with the `CovidEntry[]` type. Two types of records, the `CovidEntry[]`, which represents the `http:Created` response, and the `IsoCodeConflict`, which represents the `http:Conflict` status code response, will be used as the return types.
```

Comment:
The name doesn't matter for the payload. We have to emphasize the type, rather than the variable name.

Reply:
which represents the `http:Created` response -> which corresponds to a `201 Created` response

Reply:
Updated.